src/fairseq2/data/data_pipeline.py

```diff
@@ -88,6 +88,18 @@ def round_robin(pipelines: Sequence["DataPipeline"]) -> "DataPipelineBuilder":
                 The data pipelines to round robin.
             """
 
+        @staticmethod
+        def sample(
+            pipelines: Sequence["DataPipeline"],
+            weights: Optional[Sequence[float]] = None,
+        ) -> "DataPipelineBuilder":
+            """Extract examples from ``pipelines`` by sampling based on ``weights``.
```

Comment:
could you also document the behavior around the end of the pipeline ?
when is this pipeline considered exhausted ?